### ISSUE TITLE
Missing ethereum RPC calls make redux-devtools fail

### ISSUE THREAD

**Issue by sogoiii** [ConsenSys-archive/ganache-cli-archive#300]
A redux app that uses the `redux-devtool` chrome extension will fail to initialize. 

```javascript
bootstrap 60141db…:777 Error: Error: RPC method eth_protocolVersion not supported.
    at GethApiDouble.handleRequest (/Users/person/Documents/ethereum/github/testrpc/lib/subproviders/geth_api_double.js:58:16)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at SolcSubprovider.handleRequest (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/subproviders/solc.js:28:7)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at VmSubprovider.handleRequest (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/subproviders/vm.js:40:12)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at GethDefaults.handleRequest (/Users/person/Documents/ethereum/github/testrpc/lib/subproviders/gethdefaults.js:17:12)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at FilterSubprovider.handleRequest (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/subproviders/filters.js:87:7)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at GethApiDouble.handleRequest (/Users/person/Documents/ethereum/github/testrpc/lib/subproviders/geth_api_double.js:58:16)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at SolcSubprovider.handleRequest (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/subproviders/solc.js:28:7)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at VmSubprovider.handleRequest (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/subproviders/vm.js:40:12)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at GethDefaults.handleRequest (/Users/person/Documents/ethereum/github/testrpc/lib/subproviders/gethdefaults.js:17:12)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at FilterSubprovider.handleRequest (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/subproviders/filters.js:87:7)
    at next (/Users/person/Documents/ethereum/github/testrpc/node_modules/web3-provider-engine/index.js:95:18)
    at Object.InvalidResponse (http://192.168.145.76:3000/app.60141dbe412a57b568e1.js:18775:16)
    at RequestManager.send (http://192.168.145.76:3000/app.60141dbe412a57b568e1.js:104221:22)
    at Eth.get [as protocolVersion] (http://192.168.145.76:3000/app.60141dbe412a57b568e1.js:15727:62)
    at derez (<anonymous>:2:5795)
    at derez (<anonymous>:2:5787)
    at derez (<anonymous>:2:5787)
    at derez (<anonymous>:2:5787)
    at derez (<anonymous>:2:5787)
    at Object.decycle (<anonymous>:2:5824)
    at Object.e.stringify (<anonymous>:1:9180)
```

Redux devtool iterates over the web3.eth object and wraps all the functions. Because testrpc does not implement every rpc call, the error occurs. 

## Expected Behavior

If you have truffle, webpack, redux, and redux-devtool, the page should load and redux-devtool should be visible.

## Current Behavior

When loading the page with redux-devtools, the page outputs errors and does not render page.

## Possible Solution

From what I can see, testrpc is missing the rpc call `eth_protocolVersion` and therefor fails. 
implement the following methods to make it work:
1. eth_protocolVersion
2. personal_listAccounts
3. bzz_hive
4. bzz_info
5. ssh_version

## Steps to Reproduce (for bugs)

1. create front end project with react-redux
2. add truffle to project
3. start webpack server
4. go to page and notice how page does not render, and errors are visible in the console.

## Context

Basically, our front-end cannot use the `redux-devtools` to help facilitate development. I have to revert to console.logs rather than use a tool for debugging.

## Your Environment

* Version used:
* Environment name and version (e.g. PHP 5.4 on nginx 1.9.1):
* Server type and version:
* Operating System and version:
* Link to your project:
1. OSX el capitan
2. Chrome 60.0.3080.0
3. Node 7.9.0
4. truffle 3.2.1
5. redux-devtools chrome extension v2.14.0

**Comment by pepoospina** [comment]
This also happens with VueJs. It calls all the methods of web3 on different occasions, not only for the devtools, and it results in an error

**Comment by tcoulter** [comment]
Hi there. Thanks for submitting this issue. I'm not sure I follow exactly what's happening. The failure you posted shows that the TestRPC doesn't support `eth_protocolVersion` (which is true, but it will have a stub very soon). That said, why would `redux-devtools` expose this error? `redux-devtools` isn't affiliated with Ethereum, so it has to be your code (or some library you use) that calls this request. Is that correct?

**Comment by pepoospina** [comment]
I think is because,  sometimes,  these Javascript frameworks just call all the functions of Javascript object (`web3` in this case) for their internal workings. 

Since `web3` had the `protocolVersion` function it is being called by the framework at some point, and throws an exception. 

So, it is indeed, to some extent, not the fault of testrpc, but it would not be an issue if testrpc didn't threw an error for actual web3 methods.

**Comment by sivachaitanya** [comment]
To me  console.log('Deployed is -'+JSON.stringify( deployedecrecTest)); line throws me the error, If I doesnt use JSON.stringify, it prints out [object object]. I'm using testrpc+truffle+web3.

**Comment by sogoiii** [comment]
@tcoulter What @JoseOspina said is correct. My [pull request](https://github.com/ethereumjs/testrpc/pull/301) for a while ago worked but now has conflicts with 4.0.0. I think i have to update [ganache-core](https://github.com/trufflesuite/ganache-core) instead.

**Comment by tcoulter** [comment]
@sogoiii Don't worry about `ganache-core`. I'll get your PR in today.

**Comment by tcoulter** [comment]
This has been included in the latest version of the TestRPC. See the release notes here: https://github.com/ethereumjs/testrpc/releases

Thanks!

**Comment by zhahaoyu** [comment]
Hi I run into the same problem today. As I am trying to assign "web3" object into the state tree, redux dev tool throws like this:
```
Uncaught Error: Invalid JSON RPC response: undefined
    at Object.InvalidResponse (inpage.js:14300)
    at s.send (inpage.js:14300)
    at r.coinbase (inpage.js:14300)
    at derez (<anonymous>:2:6079)
    at derez (<anonymous>:2:6072)
    at derez (<anonymous>:2:6072)
    at derez (<anonymous>:2:6072)
    at derez (<anonymous>:2:6072)
    at derez (<anonymous>:2:6072)
    at derez (<anonymous>:2:6072)
    at derez (<anonymous>:2:6072)
    at derez (<anonymous>:2:6072)
    at derez (<anonymous>:2:6072)
    at derez (<anonymous>:2:6072)
    at Object.decycle (<anonymous>:2:6109)
    at Object.e.stringify (<anonymous>:1:9197)
    at tryCatchStringify (<anonymous>:1:18852)
    at stringify (<anonymous>:1:18960)
    at toContentScript (<anonymous>:1:20255)
    at relay (<anonymous>:1:28996)
    at <anonymous>:2:114
    at invokeFunc (<anonymous>:2:24146)
    at leadingEdge (<anonymous>:2:24223)
    at debounced (<anonymous>:2:24803)
    at Object.onMessage (<anonymous>:1:30252)
    at <anonymous>:1:21010
    at Array.forEach (<anonymous>)
    at handleMessages (<anonymous>:1:20945)
```
